streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/kstream/KStreamTest.scala

```diff
@@ -312,4 +312,28 @@ class KStreamTest extends FlatSpec with Matchers with TestDriver {
 
     testDriver.close()
   }
+
+  "join 2 KStreamToTables" should "join correctly records" in {
+    val builder = new StreamsBuilder()
+    val sourceTopic1 = "source1"
+    val sourceTopic2 = "source2"
+    val sinkTopic = "sink"
+
+    val table1 = builder.stream[String, String](sourceTopic1).toTable.groupBy((k, v) => (k, v)).count()
+    val table2 = builder.stream[String, String](sourceTopic2).toTable.groupBy((k, v) => (k, v)).count()
```

Comment:
Sorry to chime in so late on this. I looked at this test before and missed something, which I have just now noticed.

This test inserts an extra operation that defeats the purpose of the test. We should not do `groupBy().count()`, just the `toTable` on both streams, followed immediately by the join. The joiner can just concatenate both values and the verification should read the output as a map and ensure the final value is `"topic1value1topic2value1"`.

Adding the groupBy and count results in a repartition and stateful operation, potentially masking a failure to auto-materialize the toTable.

If you like, you can leave this test (there's nothing wrong with it _per se_) and just add a new test doing what I described.